src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf8Utility.Transcoding.cs

```diff
@@ -958,9 +956,18 @@
                                 goto LoopTerminatedDueToNonAsciiDataInVectorLocal;
                             }
 
-                            // narrow and write
+                            if (AdvSimd.IsSupported)
+                            {
+                                Vector64<sbyte> lower = AdvSimd.ExtractNarrowingSaturateLower(utf16Data);
+                                Vector128<sbyte> source = AdvSimd.ExtractNarrowingSaturateUpper(lower, AdvSimd.LoadVector128((short*)pInputBuffer));
+                                AdvSimd.Store((ulong*)pOutputBuffer, source.AsUInt64());
+                            }
+                            else
+                            {
+                                // narrow and write
 
```

Comment:
>ExtractNarrowingSaturateUnsignedLower [](start = 63, length = 37)

I am not much familiar with this API but just make sure that the leg that runs libraries test on arm64 (windows/linux) is run.

Reply:
@echesakovMSFT recently added these AdvSimd APIs.
I will.